### PR TITLE
first commit, reference  https://arxiv.org/pdf/1703.09252.pdf

### PR DESCRIPTION
**Pull request by acarvalh**
The process does contain special requirement to treat top in loop as massive.
The poweg input card was provided by  Alexander Grohsjean 
@OlivierBondu

### CODE REVIEW

**Comment by agrohsje** [comment]
Not sure how important it is to be consistent, but should the top mass be 172.5? 
To add. The powheg.input is a Powheg example. So in principle we should study and better understand the settings.

File: bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+!                      11  ZZ
+!                      12  gamma gamma      
+!                      -1  all decay channels closed
+
+! general parameters:
+numevts  NEVENTS     ! number of events to be generated
+ih1         1        ! hadron 1 (1 for protons, -1 for antiprotons)
+ih2         1        ! hadron 2 (1 for protons, -1 for antiprotons)
+ebeam1 7000d0        ! energy of beam 1
+ebeam2 7000d0        ! energy of beam 2
```
Comment (OlivierBondu) [inline bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input]: probably 6500 GeV per beam is enough for now :smile:

File: bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+itmx1       3        ! number of iterations for initializing the integration grid
+ncall2  10000        ! number of calls for computing the integral and finding upper bound
+itmx2       3        ! number of iterations for computing the integral and finding upper bound
+foldcsi     1        ! number of folds on csi integration
+foldy       1        ! number of folds on  y  integration
+foldphi     1        ! number of folds on phi integration
+nubound  5000        ! number of bbarra calls to setup norm of upper bounding function
+# icsimax   1        ! <= 100, number of csi subdivision when computing the upper bounds
+# iymax     1        ! <= 100, number of y subdivision when computing the upper bounds
+# xupbound 2d0       ! increase upper bound for radiation generation
```
Comment (OlivierBondu) [inline bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input]: did you play with the parameters for a smooth integration grid? These parameters seem to be the default? 
(see [this twiki](https://twiki.cern.ch/twiki/bin/viewauth/CMS/PowhegBOXPrecompiledCheckGrids))

Comment (perrozzi) [reply bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input]: usually the parameters in the example datacards in POWHEG are those used in the paper, which need to be multiplied by the number of cores...
indeed they look pretty small to me

Comment (acarvalh) [reply bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input]: @agrohsje , do you have an answer?

Comment (perrozzi) [reply bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input]: have you tried running and checking the integration grids as descripted in 
https://twiki.cern.ch/twiki/bin/viewauth/CMS/PowhegBOXPrecompiledCheckGrids

File: bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+! PARAMETERS needed for ggHH
+ptHHcut     0.001d0
+ptHHcut_CT  0.001d0
+#hdamp          150
+
+! PARAMETERS for parallel runnning
+! -- non parallel (for testing purposes only)
+iseed SEED
+#iseed    547        !  Start the random number generator with seed iseed
```
Comment (perrozzi) [inline bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input]: there are a bunch of commented parameters which would be good to remove

File: bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input (hunk 2 above)
Comment (acarvalh) [reply bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input]: Yes I did, it looks pretty smooth https://www.dropbox.com/s/w1kdnxxyp7nm82q/powegGridHH.PNG?dl=0

File: bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+! PARAMETERS needed for ggHH
+ptHHcut     0.001d0
+ptHHcut_CT  0.001d0
+
+! PARAMETERS for parallel runnning
+! -- non parallel (for testing purposes only)
+iseed SEED
+
+! -- parallel
+manyseeds 1          ! Used to perform multiple runs with different random
```
Comment (perrozzi) [inline bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input]: I am afraid this is deprecated/unused in cms. Any reason to use it?

Comment (acarvalh) [reply bin/Powheg/examples/ggHH_quark-mass-effects_withJHUGen_PDF4LHC_13TeV/ggHH.powheg.input]: No special reason, the lines were removed

**Comment by perrozzi** [comment]
looks pretty much ok to me, waiting for Olivier's last confirmation before to merge

**Comment by OlivierBondu** [comment]
looks good to me as well, thanks!

**Pull request merged**